lib/features/qr_scanner/presentation/camera_screen.dart

```diff
@@ -37,14 +37,15 @@ class _CameraScreenState extends State<CameraScreen> {
           if (state is CameraScanned) {
             log("QR code scanned: ${state.qrValue} \n Getting organisation details");
             context
-                .read<OrganisationCubit>()
+                .read<OrganisationDetailsCubit>()
                 .getOrganisationDetails(state.qrValue);
           }
         },
         builder: (context, state) {
-          return BlocConsumer<OrganisationCubit, OrganisationState>(
+          return BlocConsumer<OrganisationDetailsCubit,
+              OrganisationDetailsState>(
             listener: (context, orgState) {
-              if (orgState is OrganisationSet) {
+              if (orgState is OrganisationDetailsSetState) {
```

Comment:
Because this is a listener it only executes if the state has changed. Now if a kid scans a QR state is OrganisationDetailsSetState. They Navigate to ChooseAmountSliderScreen. If they do **not** make a donation state remains OrganisationDetailsSetState, and when they come back to make a donation this listener is not executed. They are stuck on the camera screen. 

I think this should be addressed as a separate bug and PR but I encountered it while testing the flow so I am leaving it here as well.